projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -28,58 +29,87 @@ import {
   filter,
   takeUntil,
   distinctUntilChanged,
+  catchError,
+  switchMap,
 } from 'rxjs/operators';
 
 import { Auth0ClientService } from './auth.client';
 import { WindowService } from './window';
 import { AbstractNavigator } from './abstract-navigator';
+import { connectableObservableDescriptor } from 'rxjs/internal/observable/ConnectableObservable';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService implements OnDestroy {
   private isLoadingSubject$ = new BehaviorSubject(true);
   private isAuthenticatedSubject$ = new BehaviorSubject(false);
+  private errorSubject$ = new ReplaySubject<Error>(1);
 
   // https://stackoverflow.com/a/41177163
   private ngUnsubscribe$ = new Subject();
 
+  /**
+   * Emits boolean values indicating the loading state of the SDK
+   */
   readonly isLoading$ = this.isLoadingSubject$.asObservable();
 
+  /**
+   * Emits boolean values indicating the authentication state of the user. If `true`, it means a user has authenticated.
+   * This depends on the value of `isLoading$`, so there is no need to manually check the loading state of the SDK.
+   */
   readonly isAuthenticated$ = this.isLoading$.pipe(
     filter((loading) => !loading),
     distinctUntilChanged(),
     concatMap(() => this.isAuthenticatedSubject$)
   );
 
+  /**
+   * Emits details about the authenticated user when `isAuthenticated$` is `true`.
+   */
   readonly user$ = this.isAuthenticated$.pipe(
     filter((authenticated) => authenticated),
     distinctUntilChanged(),
     concatMap(() => this.auth0Client.getUser())
   );
 
+  /**
+   * Emits errors that occur during login, or when checking for an active session on startup.
+   */
+  readonly error$ = this.errorSubject$.asObservable();
+
   constructor(
     @Inject(Auth0ClientService) private auth0Client: Auth0Client,
     @Inject(WindowService) private window: any,
     private navigator: AbstractNavigator
   ) {
     // https://github.com/angular/angular/issues/12631
     this.window = window as Window;
+
+    const checkSessionOrCallback$ = (isCallback: boolean) =>
+      iif(
+        () => isCallback,
+        this.handleRedirectCallback(),
+        defer(() => this.auth0Client.checkSession())
+      );
+
     this.shouldHandleCallback()
       .pipe(
-        concatMap((isCallback) =>
-          iif(
-            () => isCallback,
-            this.handleRedirectCallback(),
-            defer(() => this.auth0Client.checkSession())
+        switchMap((isCallback) =>
+          checkSessionOrCallback$(isCallback).pipe(
+            catchError((error) => {
+              this.errorSubject$.next(error);
+              this.navigator.navigateByUrl('/');
+              return of(undefined);
+            })
```

Comment:
Normally when an error occurs, the observable is completed. Here, I'd like it to continue on to the end, so I use a switchMap to create a new observable which tracks any errors that occur into `error$` then throws the observable away, meaning the flow can continue on.

I also navigate the browser away to remove the querystring. @lbalmaceda, would appreciate your thoughts here; this might be too naive an approach.

Reply:
I saw this catch approach [here](https://stackoverflow.com/a/53554276), I think that's fine. Do we need to return `undefined` or could we just do `of()`? 

For the redirect, I think we're already doing something like that for when we successfully handle the callback. What do you think about moving the `navigateByUrl` line to some "resetQuery" method and pipe that in both places as the final step?

And something to think about regarding that approach. If the "callback URL" that the user set in the options is different from the app's root. Would it be OK to take them to the root when this fails?

Reply:
> If the "callback URL" that the user set in the options is different from the app's root. Would it be OK to take them to the root when this fails?

Hmm good point. I'll make this a bit smarter and just make it so the query params are removed, rather than just redirecting to root.

Reply:
> Do we need to return undefined or could we just do of()?

It turns out `of()` and `of(undefined)` are not equivalent. The former will complete the observable immediately and not continue processing, whereas `of(undefined)` will emit `undefined` (as `undefined` is itself a value) and continue on down the chain, which is the desired result.